drivers/at/at.c

```diff
@@ -0,0 +1,248 @@
+/*
+ * Copyright (C) 2017 Kaspar Schleiser <kaspar@schleiser.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+#include <errno.h>
+#include <string.h>
+
+#include "at.h"
+#include "fmt.h"
+#include "isrpipe.h"
+#include "periph/uart.h"
+#include "xtimer.h"
+
+#define ENABLE_DEBUG (0)
+#include "debug.h"
+
+#ifndef AT_PRINT_INCOMING
+#define AT_PRINT_INCOMING (0)
+#endif
+
+int at_dev_init(at_dev_t *dev, uart_t uart, uint32_t baudrate, char *buf, size_t bufsize)
+{
+    dev->uart = uart;
+    isrpipe_init(&dev->isrpipe, buf, bufsize);
+    uart_init(uart, baudrate, (uart_rx_cb_t) isrpipe_write_one,
+              &dev->isrpipe);
+
+    return 0;
+}
+
+int at_expect_bytes(at_dev_t *dev, const char *bytes, uint32_t timeout)
+{
+    while (*bytes) {
+        char c;
+        int res;
+        if ((res = isrpipe_read_timeout(&dev->isrpipe, &c, 1, timeout)) == 1) {
+            if (AT_PRINT_INCOMING) {
+                print(&c, 1);
+            }
+            if (c != *bytes++) {
+                return -1;
+            }
+        }
+        else {
+            return res;
+        }
+    }
+
+    return 0;
+}
+
+void at_send_bytes(at_dev_t *dev, const char *bytes, size_t len)
+{
+    uart_write(dev->uart, (const uint8_t *)bytes, len);
+}
+
+int at_send_cmd(at_dev_t *dev, const char *command, uint32_t timeout)
+{
+    size_t cmdlen = strlen(command);
+
+    uart_write(dev->uart, (const uint8_t *)command, cmdlen);
+    uart_write(dev->uart, (const uint8_t *)AT_SEND_EOL, AT_SEND_EOL_LEN);
+
+    if (AT_SEND_ECHO) {
```

Comment:
@vincent-d @kaspar030, here is the change that allows to skip the echo check at build time.